eZ/Publish/Core/Repository/LocationResolver/PermissionAwareLocationResolver.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Repository\LocationResolver;
+
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Location;
+
+final class PermissionAwareLocationResolver implements LocationResolver
+{
+    /** @var \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    public function __construct(LocationService $locationService)
+    {
+        $this->locationService = $locationService;
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     */
+    public function resolveLocation(ContentInfo $contentInfo): Location
+    {
+        try {
+            $location = $this->locationService->loadLocation($contentInfo->mainLocationId);
```

Comment:
What if `$contentInfo->mainLocationId` is `null`? In 7.5 this will throw NotFound, but because it's gonna be cast to 0. `ezplatform-kernel` however should yield a Fatal error, I expect.

See how the entire PR merges up to possibly uncover more issues.

Reply:
Yes, when `mainLocationId` is null in `ezplatform-kernel` than we would have a fatal error as `loadLocation` expects an `int`. 

I've added another validation for such a situation: https://github.com/ezsystems/ezpublish-kernel/pull/3050/files#diff-4475afc98a247f7b03015b9402086e29R39